development/extensions/index.md

```diff
@@ -24,51 +24,125 @@ One of the primary aims of BlueOS is to be a platform that is readily extendable
 
 Accordingly, BlueOS has been designed with a containerised architecture, so the core functionality is kept separate from Extensions, and each can be distributed and updated independently of the other. The built in Extension system allows developers and users to find and install add-on software packages to BlueOS, and manage updates, permissions, and resource limits of those Extensions through the [Extensions Manager](../../advanced-usage/#extensions-manager).
 
-For some additional context and discussion, please see [this forum thread](https://discuss.bluerobotics.com/t/external-integrations-extensions/10912) about the BlueOS Extensions architecture. Feel free to contribute ideas or ask questions there as relevant.
-
 ## Implementation
 
 {% warning() %}
 The BlueOS Extensions system is currently still in a `beta` stage of development, so some implementation changes are expected in the near future.
 {% end %}
 
+The BlueOS Extensions system consists of three major components:
+
+1. **Extension Packages**, which are installable applications (Extensions) that run on the BlueOS computer, alongside [BlueOS-core](../core/) and [BlueOS-bootstrap](../bootstrap/)
+1. **BlueOS Bazaar**, an online web store that allows BlueOS users to find and install Extension Packages, and provides a standardised place for developers to offer Extensions through
+   - Anyone can see which Extensions are available, so (for example):
```

Comment:
As far as I'm aware that's not in a shareable form yet - these docs are for the current state of development.

The Extension registration process will also change significantly once the real Bazaar is up and running.

Reply:
Should we add https://docs.bluerobotics.com/BlueOS-Extensions-Repository/ ?

Reply:
I was originally not wanting to, because it's not really the Bazaar, but I guess conceptually it is part of the current equivalent of the Bazaar, so I've made it work.